src/dali/plugin/input/rest/coinbase_pro.py

```diff
@@ -354,10 +354,10 @@ def _process_fills(self, transaction: Any, in_transaction_list: List[InTransacti
                     transaction_type="Buy",
                     spot_price=str(to_currency_price),
                     crypto_in=str(to_currency_size),
-                    crypto_fee=str(crypto_fee),
-                    fiat_in_no_fee=None,
+                    crypto_fee=None,
+                    fiat_in_no_fee=str(usd_volume),
                     fiat_in_with_fee=None,
-                    fiat_fee=None,
+                    fiat_fee=str(from_currency_price * crypto_fee),
```

Comment:
A couple of questions (I'm shooting in the dark, because I don't have a buy-side fill log to look at):
- is fiat_in_no_fee == crypto_in * spot_price)? If it is you can leave fiat_in_no_fee=None and RP2 will compute it automatically (https://github.com/eprbell/rp2/blob/main/src/rp2/in_transaction.py#L77).
- also a bit curious why you need to set the fiat_fee, because it looks to me like you're paying the fee twice (once as crypto_fee in the OutTransaction and once here in the InTransaction).

Reply:
> also a bit curious why you need to set the fiat_fee, because it looks to me like you're paying the fee twice (once as crypto_fee in the OutTransaction and once here in the InTransaction).

The fee is only paid once, I am setting the fee in in transaction so that the fee will be taken into account when calculating the cost basis in future sales. Technically if we set `crypto_fee="0"` for the out transaction, we will still get the same tax calculation. But since in reality the transaction fee was paid using the from_currency (btc in my example) in coinbase pro, i figured it would be more accurate to record it as crypto fee as part of the out transaction.

Reply:
When you set crypto_fee in the OutTransaction, RP2 will automatically convert it to fiat and populate fiat_fee (https://github.com/eprbell/rp2/blob/main/src/rp2/in_transaction.py#L66). The fiat_fee is taken into account and used to lower proceeds for the OutTransaction (see https://github.com/eprbell/rp2/issues/6): you get the tax benefit in this way. This is why I'm thinking there should be no need to set fiat_fee in the InTransaction and to raise the cost basis (because the fee was already used to lower the proceeds in the OutTransaction). I think this code might both lower your proceeds in the OutTransaction and raise the cost basis in the InTransaction, thus giving a double deduction.

Reply:
So after some more thinking over this, I think the correct implementation would be:
* line 315 (inside `if fill_side == _SELL`), add two lines:
```
out_crypto_fee: RP2Decimal = ZERO
in_crypto_fee: RP2Decimal =  crypto_fee
```
* line 321 (inside `if fill_side == _BUY`), add two lines:
```
out_crypto_fee: RP2Decimal = crypto_fee
in_crypto_fee: RP2Decimal =  ZERO
```
And then assign `out_crypto_fee` and `in_crypto_fee` to the crypto_fee field of the Out/InTransaction. There is no need to modify fiat_fee.

Reply:
> I think this code might both lower your proceeds in the OutTransaction and raise the cost basis in the InTransaction, thus giving a double deduction.

This was my intention as noted in my PR description :) Since this is not well defined in the IRS code, I figured we could count it as fees for both the buy side and sell side.

But I agree with you that it it would be safer to only count the fee on one side of the trade. Will update the code with a test case for buy side.